### PR TITLE
feature(prow-jobs): support run CI for feature branches created on release versions

### PR DESCRIPTION
**Pull request by wuhuizuo**
they are one of these format:
- feature/release-X.Y-*
- feature/release-X.Y.Z-*

Signed-off-by: wuhuizuo <wuhuizuo@126.com>

### CODE REVIEW

File: prow-jobs/tikv/pd/latest-presubmits.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# struct ref: https://pkg.go.dev/k8s.io/test-infra/prow/config#Presubmit
+# struct ref: https://pkg.go.dev/sigs.k8s.io/prow/pkg/config#Presubmit
```
Comment (purelind) [inline prow-jobs/tikv/pd/latest-presubmits.yaml:1]: Need ignore branch that match `feature/release-x.y` in this file

File: prow-jobs/tikv/tikv/latest-presubmits.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# struct ref: https://pkg.go.dev/k8s.io/test-infra/prow/config#Presubmit
+# struct ref: https://pkg.go.dev/sigs.k8s.io/prow/pkg/config#Presubmit
```
Comment (purelind) [inline prow-jobs/tikv/tikv/latest-presubmits.yaml:1]: Need ignore branch that match `feature/release-x.y` in this file

File: prow-jobs/tikv/tikv/release-7.5-presubmits.yaml [hunk 3]
```diff
@@ -1,8 +1,9 @@
-# struct ref: https://pkg.go.dev/k8s.io/test-infra/prow/config#Presubmit
 global_definitions:
   branches: &branches
     - ^release-7\.5(\.\d+)?(-\d+)?(-v[\.\d]+)?$
+    - ^feature/release-7\.5(.\d+)?-.+$ # for feature branch based on v7.5.x
```
Comment (wuhuizuo) [inline prow-jobs/tikv/tikv/release-7.5-presubmits.yaml:4]: key chagnes

File: prow-jobs/pingcap/tidb/latest-presubmits.yaml [hunk 4]
```diff
@@ -1,78 +1,81 @@
-# struct ref: https://pkg.go.dev/k8s.io/test-infra/prow/config#Presubmit
+global_definitions:
+  brancher: &brancher
+    branches:
+      - ^master$
+      - ^feature/.+
+    skip_branches: # skip feature branches based on released version.
+      - ^feature/release-\d+\.\d+(.\d+)?-.+$
```
Comment (wuhuizuo) [inline prow-jobs/pingcap/tidb/latest-presubmits.yaml:7]: key changes

File: prow-jobs/pingcap/tidb/release-7.5-presubmits.yaml [hunk 5]
```diff
@@ -1,4 +1,9 @@
-# struct ref: https://pkg.go.dev/k8s.io/test-infra/prow/config#Presubmit
+global_definitions:
+  branches: &branches
+    - ^release-7\.5(\.\d+)?(-\d+)?(-v[\.\d]+)?$
+    - ^feature/release-7\.5(.\d+)?-.+$ # for feature branch based on v7.5.x
```
Comment (wuhuizuo) [inline prow-jobs/pingcap/tidb/release-7.5-presubmits.yaml:4]: key changes

File: prow-jobs/pingcap/tiflow/latest-presubmits.yaml [hunk 6]
```diff
@@ -37,11 +40,11 @@ presubmits:
       run_before_merge: true
       context: pull-cdc-integration-storage-test
       skip_report: false
-      trigger: "(?m)^/test (?:.*? )?(cdc-integration-storage-test|all)(?: .*?)?$"
+      trigger:
+        "(?m)^/test (?:.*? )?(cdc-integration-storage-test|all)(?: .*?)?$"
```
[the remark this answers is not in the document]
Comment (wuhuizuo) [reply prow-jobs/pingcap/tiflow/latest-presubmits.yaml:44]: yaml format, i will revert it.

**Pull request merged**